src/run.rs

```diff
@@ -37,9 +38,23 @@ pub fn run(
     while context.quit == QuitType::DoNot {
         backend.render(TuiView::new(context));
 
-        // compute preview render sizes
-        if let Ok(area) = backend.terminal_ref().size() {
-            render_result = process_preview_on_change(&context, area, render_result);
+        {
+            let config = context.config_ref();
+            let preview_options = config.preview_options_ref();
+            if let Some(shown_hook) = preview_options.preview_shown_hook_script.as_ref() {
```

Comment:
This handling has one downside: The shown-hood is only invoked if also the remove-hook is configured (and vice versa). While this is completely fine for the Überzug use case, this might be unexpected for users who have a different idea of what intersting stuff could be done with these hooks. If some user just wants to pop up a desktop notification with an image or whatever else, he won't need the remove-hook, but still needs to define one, just to get the shown hook-working. This might be a stupid example, but people might come up with more creative ideas of what can be done with it.

Reply:
Yeah, I can split it up so they aren't dependent on each other :+1: